octopusdeploy_framework/resource_channel.go

```diff
@@ -0,0 +1,275 @@
+
+	channel := channels.NewChannel(channelName, projectId)
+
+	channel.ID = model.ID.ValueString()
+	channel.Description = model.Description.ValueString()
+	channel.IsDefault = model.IsDefault.ValueBool()
+	channel.LifecycleID = model.LifecycleId.ValueString()
+	channel.Rules = expandChannelRules(model.Rule)
+	channel.SpaceID = model.SpaceId.ValueString()
+	channel.TenantTags = expandStringList(model.TenantTags)
+
+	return channel
+}
+
+func expandChannelRules(rules types.List) []channels.ChannelRule {
+	if rules.IsNull() || rules.IsUnknown() {
+		return nil
+	}
+
+	var rulesMap []map[string]interface{}
+	rules.ElementsAs(context.Background(), &rulesMap, false)
+
+	var channelRules []channels.ChannelRule
+	for _, ruleMap := range rulesMap {
+		channelRule := expandChannelRule(ruleMap)
+		channelRules = append(channelRules, channelRule)
+	}
+
+	return channelRules
+}
+
+func expandChannelRule(rule map[string]interface{}) channels.ChannelRule {
+	var channelRule channels.ChannelRule
+
+	channelRule.ID = rule["id"].(string)
+	actionPackage := rule["action_package"].([]map[string]interface{})
+	channelRule.ActionPackages = expandChannelRuleDeploymentActionPackages(actionPackage)
+	channelRule.Tag = rule["tag"].(string)
+	channelRule.VersionRange = rule["version_range"].(string)
+
+	return channelRule
+}
+
+func expandChannelRuleDeploymentActionPackages(actionPackages []map[string]interface{}) []packages.DeploymentActionPackage {
+	var actionPackagesExpanded []packages.DeploymentActionPackage
+	for _, actionPackage := range actionPackages {
+		actionPackageExpanded := expandChannelRuleDeploymentActionPackage(actionPackage)
+		actionPackagesExpanded = append(actionPackagesExpanded, actionPackageExpanded)
+	}
+	return actionPackagesExpanded
+}
+
+func expandChannelRuleDeploymentActionPackage(actionPackageMap map[string]interface{}) packages.DeploymentActionPackage {
+	return packages.DeploymentActionPackage{
+		DeploymentAction: actionPackageMap["deployment_action"].(string),
+		PackageReference: actionPackageMap["package_reference"].(string),
+	}
+
+}
+
+func flattenChannel(ctx context.Context, channel *channels.Channel, model schemas.ChannelModel) schemas.ChannelModel {
+	model.ID = types.StringValue(channel.GetID())
+	model.Description = types.StringValue(channel.Description)
+
+	if !channel.IsDefault && model.IsDefault.IsNull() {
+		model.IsDefault = types.BoolNull()
+	} else {
+		model.IsDefault = types.BoolValue(channel.IsDefault)
+	}
+
+	if channel.LifecycleID == "" && model.LifecycleId.IsNull() {
+		model.LifecycleId = types.StringNull()
+	} else {
+		model.LifecycleId = types.StringValue(channel.LifecycleID)
+	}
+
+	model.Name = types.StringValue(channel.Name)
+	model.ProjectId = types.StringValue(channel.ProjectID)
+
+	model.Rule = flattenChannelRules(channel.Rules, model.Rule)
+
+	if channel.SpaceID == "" && model.SpaceId.IsNull() {
+		model.SpaceId = types.StringNull()
+	} else {
+		model.SpaceId = types.StringValue(channel.SpaceID)
+	}
```

Comment:
```suggestion
        model.SpaceId = types.StringValue(channel.SpaceID)
```

Reply:
I applied these changes, but it broke state consistency. If I remember correctly, this is required in order to keep the state consistent with the API response.

Reply:
🤔 that's weird since we don't have that issue in other resources...anywho, if it causes grief then just leave it as you had it